internal/provider/waypoint/data_source_waypoint_add_on.go

```diff
@@ -269,11 +311,40 @@ func (d *DataSourceAddOn) Read(ctx context.Context, req datasource.ReadRequest,
 		}
 	}
 
-	diags = readOutputs(ctx, addOn, &state)
+	ol := readOutputs(addOn.OutputValues)
+	if len(ol) > 0 {
+		state.OutputValues, diags = types.ListValueFrom(ctx, types.ObjectType{AttrTypes: outputValue{}.attrTypes()}, ol)
+	} else {
+		state.OutputValues = types.ListNull(types.ObjectType{AttrTypes: outputValue{}.attrTypes()})
+	}
```

Comment:
Do you think you could explain the change here a little bit? It looks like this is to make this more readable but I want to better understand the why of this change.

Reply:
It simplifies the function to only operate on the common thing, the outputs, so we can reuse the method for both resources and data sources.